lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+ 
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @registered_vehicles << vehicle
```

Comment:
We only want to register a vehicle if it's an included service, so you'll want to wrap this logic into an if statement checking that it's available. 

Nice use of a helper method in the Vehicle class!
You might want to consider a helper method for collecting the fees as well. Because this registration method can get pretty complex, you could create a helper method in this class that collects the fees based on the vehicle type. 

Finally -- be mindful of return values in the interaction pattern - we want this method to either return the @registered_vehicles attribute (if successful) or return false if vehicle registration is not a provided service